lib/copyright.js

```diff
@@ -0,0 +1,171 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: strong-tools
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+var Project = require('./project');
+var Promise = require('bluebird');
+var _ = require('lodash');
+var git = require('./git');
+var paths = require('./paths');
+var readFile = Promise.promisify(require('fs').readFile);
+var writeFile = Promise.promisify(require('fs').writeFile);
+
+// Components of the copyright header.
+var COPYRIGHT = [
+  'Copyright <%= owner %> <%= years %>. All Rights Reserved.',
+  'Node module: <%= name %>',
+];
+var LICENSE = [
+  'This file is licensed under the <%= license %>.',
+  'License text available <%= ref %>',
+];
+var CUSTOM_LICENSE = [];
+if (process.env.SLT_COPYRIGHT) {
+  CUSTOM_LICENSE = require(process.env.SLT_COPYRIGHT);
+}
+
+// Compiled templates for generating copyright headers
+var UNLICENSED = _.template(COPYRIGHT.join('\n'));
+var LICENSED = _.template(COPYRIGHT.concat(LICENSE).join('\n'));
+var CUSTOM = UNLICENSED;
+if (CUSTOM_LICENSE.length) {
+  CUSTOM = _.template(COPYRIGHT.concat(CUSTOM_LICENSE).join('\n'));
+}
+
+// Patterns for matching previously generated copyright headers
+var BLANK = /^\s*$/;
+var ANY = COPYRIGHT.concat(LICENSE, CUSTOM_LICENSE).map(function(l) {
+  return new RegExp(l.replace(/<%[^>]+%>/g, '.*'));
```

Comment:
This turns templates into regex that will match most renderings of it:
`'This is an ejs style template with <%= data %> in it'`
`/This is an ejs style template with .* in it/g`